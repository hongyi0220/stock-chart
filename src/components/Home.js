import React from 'react';
import PropTypes from 'prop-types';
import Sidebar from './Sidebar';

export default class Home extends React.Component {
  static propTypes = {
    history: PropTypes.shape({
      push: PropTypes.func,
      location: PropTypes.shape({
        pathname: PropTypes.string
      })
    }),
    isSidebarOn: PropTypes.bool,
    toggleSidebar: PropTypes.func,
  };

  render() {
    return (
      <div className='home-page-container'>
        <div className='logo-wrapper'>
          <img onClick={() => {this.props.history.push('/');}} src='/img/logo2.png'/>
        </div>
        <div onClick={()=> {this.props.history.push('/chart');}} className='view-chart-container'>
          <img src='/img/line-chart2.png'/>
          <div className='view-chart-text-wrapper'>View Chart</div>
        </div>
        <Sidebar isSidebarOn={this.props.isSidebarOn}/>
        {!this.props.isSidebarOn && <div className='arrow' onClick={this.props.openSidebar}>></div>}{/* eslint-disable-line*/}
        <div className='footer'>
          <a className='footer-item' href='https://hongyi0220.github.io'>yung</a>
          <hr />
          <a className='footer-item' href='https://github.com/hongyi0220/stock-chart'>repo</a>
        </div>
      </div>
    );
  }
}
