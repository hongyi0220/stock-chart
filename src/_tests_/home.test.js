import React from 'react';
import Home from '../components/Home';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow, render, mount } from 'enzyme';

Enzyme.configure({ adapter: new Adapter() });

test('Home renders', () => {
  const wrapper = mount(<Home isSidebarOn={false} />);

  expect(wrapper).toMatchSnapshot();
});
