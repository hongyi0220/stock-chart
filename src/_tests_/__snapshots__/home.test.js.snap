// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders 1`] = `
<Home
  isSidebarOn={false}
>
  <div
    className="home-page-container"
  >
    <div
      className="logo-wrapper"
    >
      <img
        onClick={[Function]}
        src="/img/logo2.png"
      />
    </div>
    <div
      className="view-chart-container"
      onClick={[Function]}
    >
      <img
        src="/img/line-chart2.png"
      />
      <div
        className="view-chart-text-wrapper"
      >
        View Chart
      </div>
    </div>
    <Sidebar
      isSidebarOn={false}
    >
      <Sidebar
        animation="overlay"
        direction="left"
        visible={false}
        width="wide"
      >
        <div
          className="ui overlay left wide sidebar"
        >
          <div
            className="about-this-app-container"
          >
            <div
              className="about-text-wrapper"
            >
              About This App
            </div>
            <div
              className="tech-text-wrapper"
            >
              Front-end tech stack
            </div>
            <div
              className="front-end logos-container"
            >
              <div
                className="react logo-container"
              >
                <img
                  src="/img/logos/react-logo.png"
                />
                <div
                  className="logo-text-wrapper"
                >
                  React.js
                </div>
              </div>
              <div
                className="semantic logo-container"
              >
                <img
                  src="/img/logos/semantic-ui-logo.png"
                />
                <div
                  className="logo-text-wrapper"
                >
                  Semantic-UI-React
                </div>
              </div>
            </div>
            <div
              className="tech-text-wrapper"
            >
              Back-end tech stack
            </div>
            <div
              className="back-end logos-container"
            >
              <div
                className="nodejs logo-container"
              >
                <img
                  src="/img/logos/nodejs-logo2.png"
                />
                <div
                  className="logo-text-wrapper"
                >
                  Node.js
                </div>
              </div>
              <div
                className="express logo-container"
              >
                <img
                  src="/img/logos/expressjs-logo2.png"
                />
                <div
                  className="logo-text-wrapper"
                >
                  Express.js
                </div>
              </div>
              <div
                className="mongodb logo-container"
              >
                <img
                  src="/img/logos/mongodb-logo.png"
                />
                <div
                  className="logo-text-wrapper"
                >
                  MongoDB
                </div>
              </div>
            </div>
            <div
              className="tech-text-wrapper"
            >
              API
            </div>
            <div
              className="api logos-container"
            >
              <div
                className="quandl logo-container"
              >
                <img
                  src="/img/logos/quandl-logo.png"
                />
                <div
                  className="logo-text-wrapper"
                >
                  Quandl
                </div>
              </div>
            </div>
            <div
              className="tech-text-wrapper"
            >
              Key modules
            </div>
            <div
              className="modules logos-container"
            >
              <div
                className="socketio logo-container"
              >
                <img
                  src="/img/logos/socketio-logo.gif"
                />
                <div
                  className="logo-text-wrapper"
                >
                  Socket.io
                </div>
              </div>
              <div
                className="highcharts logo-container"
              >
                <img
                  src="/img/logos/highcharts-logo.png"
                />
                <div
                  className="logo-text-wrapper"
                >
                  Highcharts
                </div>
              </div>
            </div>
            <div>
              Logo made with
              <a
                href="https://www.designevo.com/"
                title="Free Online Logo Maker"
              >
                DesignEvo
              </a>
            </div>
          </div>
        </div>
      </Sidebar>
    </Sidebar>
    <div
      className="arrow"
    >
      &gt;
    </div>
    <div
      className="footer"
    >
      <a
        className="footer-item"
        href="https://hongyi0220.github.io"
      >
        yung
      </a>
      <hr />
      <a
        className="footer-item"
        href="https://github.com/hongyi0220/stock-chart"
      >
        repo
      </a>
    </div>
  </div>
</Home>
`;
